src/NServiceBus.RavenDB/SagaPersister/SagaPersister.cs

```diff
@@ -19,32 +19,37 @@ public async Task Save(IContainSagaData sagaData, SagaCorrelationProperty correl
                 return;
             }
 
-            var container = new SagaDataContainer
-            {
-                Id = DocumentIdForSagaData(documentSession, sagaData),
-                Data = sagaData
-            };
-
             if (correlationProperty == null)
             {
                 return;
             }
 
-            container.IdentityDocId = SagaUniqueIdentity.FormatId(sagaData.GetType(), correlationProperty.Name, correlationProperty.Value);
-
+            var container = new SagaDataContainer
+            {
+                Id = DocumentIdForSagaData(documentSession, sagaData),
+                Data = sagaData,
+                IdentityDocId = SagaUniqueIdentity.FormatId(sagaData.GetType(), correlationProperty.Name, correlationProperty.Value)
+            };
             await documentSession.StoreAsync(container, string.Empty, container.Id).ConfigureAwait(false);
-            await documentSession.StoreAsync(new SagaUniqueIdentity
+            documentSession.StoreVersionInMetadata(container);
+            var sagaUniqueIdentity = new SagaUniqueIdentity
             {
                 Id = container.IdentityDocId,
                 SagaId = sagaData.Id,
                 UniqueValue = correlationProperty.Value,
                 SagaDocId = container.Id
-            }, changeVector: string.Empty, id: container.IdentityDocId).ConfigureAwait(false);
+            };
+            await documentSession.StoreAsync(sagaUniqueIdentity, changeVector: string.Empty, id: container.IdentityDocId).ConfigureAwait(false);
+            documentSession.StoreVersionInMetadata(sagaUniqueIdentity);
         }
 
         public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session, ContextBag context)
         {
-            //no-op since the dirty tracking will handle the update for us
+            // in case we update the saga with a new persister
+            var container = context.Get<SagaDataContainer>($"{SagaContainerContextKeyPrefix}{sagaData.Id}");
```

Comment:
Previously this was No Op. Now we need to at least store the metadata because we could be loading a saga from a previous version and we want to make sure the metadata reflects the version of the persister when the document was last saved.